plugins/prometheus_plugin/metrics.hpp

```diff
@@ -116,6 +135,35 @@ struct catalog_type {
    void update(const net_plugin::p2p_connections_metrics& metrics) {
       num_peers.Set(metrics.num_peers);
       num_clients.Set(metrics.num_clients);
+      for(size_t i = 0; i < metrics.stats.peers.size(); ++i) {
+         auto addr = boost::asio::ip::make_address_v6(metrics.stats.peers[i].address).to_string();
+         boost::replace_all(addr, ":", "_COLON_");
+         boost::replace_all(addr, ".", "_DOT_");
+         addr.insert(0, "ip_");
+         addr.append("_");
+         addr.append(to_string(metrics.stats.peers[i].port));
+         addresses.push_back(addr);
+         auto add_and_set_gauge = [&](std::string label_value, 
+                                      std::vector<std::reference_wrapper<Gauge>>& gauges,
+                                      auto value) {
```

Comment:
`const auto& value`. I guess value can only be a `double`, so maybe just use `double`.

Reply:
It can also be `bool`.